src/lib/rootfs/rootfs.c

```diff
@@ -245,18 +254,22 @@ int singularity_rootfs_check(void) {
 int singularity_rootfs_chroot(void) {
     
     singularity_priv_escalate();
-    singularity_message(VERBOSE, "Entering container file system root: %s\n", joinpath(mount_point, OVERLAY_FINAL));
-    if ( chroot(joinpath(mount_point, OVERLAY_FINAL)) < 0 ) { // Flawfinder: ignore (yep, yep, yep... we know!)
-        singularity_message(ERROR, "failed enter container at: %s\n", joinpath(mount_point, OVERLAY_FINAL));
+
+    if ( fchdir(singularity_rootfs_fd()) < 0 ) {
```

Comment:
I am not technically fixing anything, however it is theoretically more secure to first fchdir to the rootfs fd, and then chroot on the current directory. We don't have to worry about issues with path name strings, changing links, etc.. because of doing it this way I believe.

And if we need to chdir after the chroot, then we can just add one last call to chdir("/") after the chroot